pkg/maps/policymap/policymap.go

```diff
@@ -453,7 +455,15 @@ func newMap(path string) *PolicyMap {
 // protected by this map.
 func OpenOrCreate(path string) (*PolicyMap, error) {
 	m := newMap(path)
+
 	err := m.OpenOrCreate()
+	if errors.Is(err, ebpf.ErrMapIncompatible) {
+		err = m.Recreate()
```

Comment:
Shouldn't this be part of the previous commit?

Reply:
This commit has been removed, but the goal was to split off the change to the building block from the change to the consumer, as it might be easier to review that way.